src/osemgrep/cli_scan/Scan_CLI.ml

```diff
@@ -876,25 +872,39 @@ let cmdline_term ~allow_empty_config : conf Term.t =
       | _else_ -> default.output_format
     in
     let engine_type =
-      match (oss, pro_lang, pro_intrafile, pro) with
-      | false, false, false, false when secrets ->
-          Engine_type.(PRO Language_only)
-      | false, false, false, false -> default.engine_type
+      (* This first bit just rules out mutually exclusive options. *)
+      (match (oss, pro_lang, pro_intrafile, pro) with
       | true, false, false, false when secrets ->
           Error.abort
             "Mutually exclusive options --oss/--beta-testing-secrets-enabled"
-      | true, false, false, false -> OSS
-      | false, true, false, false -> PRO Engine_type.Language_only
-      | false, false, true, false -> PRO Engine_type.Intrafile
-      | false, false, false, true -> PRO Engine_type.Interfile
+      | false, false, false, false
+      | true, false, false, false
+      | false, true, false, false
+      | false, false, true, false
+      | false, false, false, true ->
+          ()
       | _else_ ->
           (* TOPORT: list the possibilities *)
           Error.abort
             "Mutually exclusive options \
-             --oss/--pro-languages/--pro-intrafile/--pro"
+             --oss/--pro-languages/--pro-intrafile/--pro");
+      (* Now select the engine type *)
+      if oss then Engine_type.OSS
+      else
+        let analysis =
+          match () with
+          | _ when pro_intrafile -> Engine_type.Deep_intrafile
+          | _ when pro -> Engine_type.Deep_interfile
+          | _ -> Engine_type.Intrafile
```

Comment:
What is the difference between Deep_intrafile and Intrafile @IagoAbal ?

Reply:
This might be better described as intraprocedural but `Intrafile` is just refering to the OSS scan type.